src/api/buildData.ts

```diff
@@ -80,27 +96,47 @@ const getUserData = async (projectID: string, user: UserProfile, api: Api) => {
 export const buildData = async (
   projectID: string,
   manuscriptID: string,
-  doc: ManuscriptNode,
+  updateState: (state: Partial<state>) => void,
   api: Api
 ) => {
   const user = await api.getUser()
   if (!user) {
     return {}
   }
 
-  const manuscriptNode = doc as ActualManuscriptNode
-  const state = await getManuscriptData(api, manuscriptNode.attrs.prototype)
+  const doc = await getDocumentData(projectID, manuscriptID, api)
+  const state = await getManuscriptData(doc.doc.attrs.prototype, api)
   const [project, manuscript] = await getIdModels(api, projectID, manuscriptID)
   const role = project ? getUserRole(project, user.userID) : null
   const users = await getUserData(projectID, user, api)
 
+  const stepsExchanger = new StepsExchanger(
+    projectID,
+    manuscriptID,
+    doc.version,
+    api
+  )
+  stepsExchanger.isThrottling.onChange((value: boolean) => {
```

Comment:
Do you think we can make it work without putting stepExchanger into the store? This looks like something what shouldn't be generally accessible to components since it's designed specifically for editor props and works directly with api. When I created it, I wanted to have no reference to it in the article editor on purpose to avoid potential misuse.

Reply:
I can do that, but it would require storing the version to the store, and somehow get access to the api. I can just create a new `Api` instance if that's acceptable.

Reply:
I think we can either create a new api instance (we can singleton it if it's not safe for WS listeners) but I'm not sure why we need to store the current version in the store. As far as I remember the only reason we need to set the current version is to initiate collab plugin on when we load/reload the editor and that initial version value you already add to the store by unpacking output of getDocumentData into the store on line 136 in this file (probably would be better to put it into the store with a label like 'initialVersion' or similar).

Reply:
I made more changes and got the stepsExchanger out of the source and created it in `useCreateEditor` instead.